komga/src/main/kotlin/org/gotson/komga/interfaces/rest/BookController.kt

```diff
@@ -550,6 +551,25 @@ class BookController(
     }
   }
 
+  @DeleteMapping("api/v1/books/{bookId}")
+  @PreAuthorize("hasRole('$ROLE_ADMIN')")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  fun deleteBook(
+    @PathVariable bookId: String,
+    @AuthenticationPrincipal principal: KomgaPrincipal
+  ) {
+    bookRepository.findByIdOrNull(bookId)?.let { book ->
+      if (!principal.user.canAccessBook(book)) throw ResponseStatusException(HttpStatus.FORBIDDEN)
+      if (!Files.isWritable(book.path)) {
+        logger.warn { "Cannot delete ${book.path}: Permission denied" }
+        throw ResponseStatusException(HttpStatus.CONFLICT)
+      }
+
+      bookLifecycle.deleteOne(book)
+      taskReceiver.deleteFile(book.url)
```

Comment:
The first operation is synchronous, while the second is asynchronous, which could lead to consistency issues.

There are also side effects when deleting a book, which are handled by the scan, for instance the reordering of the remaining books in a series, or the series metadata aggregation.

We should also consider deleting the book's sidecar files.